packages/editor/src/client/panel/mapLeafletEditPanel.tsx

```diff
@@ -0,0 +1,213 @@
+          onChange={items => setItems(items)}
+          defaultValue={defaultItemValue}>
+          {props => <MapLeafletItems {...props} />}
+        </ListInput>
+      </PanelSection>
+    </Panel>
+  )
+}
+
+export function MapLeafletItems({value, onChange}: FieldProps<MapLeafletItem>) {
+  const {address, lat, lng, title, description} = value
+
+  return (
+    <>
+      <Box display="flex" flexDirection="column">
+        <AddressInput
+          label="Address"
+          value={[{address: address, lat: lat, lng: lng}]}
+          marginBottom={Spacing.ExtraSmall}
+          onChange={markerPoints => {
+            if (markerPoints !== undefined && markerPoints.length > 0) {
+              onChange({
+                ...value,
+                address: markerPoints[1].address,
+                lat: markerPoints[1].lat,
+                lng: markerPoints[1].lng
+              })
+            }
+          }}
+        />
+        <TextInput
+          marginBottom={Spacing.ExtraSmall}
+          type="number"
+          label="Latitude"
+          value={lat}
+          onChange={e => onChange({...value, lat: parseInt(e.target.value)})}
+          required
+        />
+        <TextInput
+          marginBottom={Spacing.ExtraSmall}
+          type="number"
+          label="Longitude"
+          value={lng}
+          onChange={e => onChange({...value, lng: parseInt(e.target.value)})}
+          required
+        />
+        <TextInput
+          marginBottom={Spacing.ExtraSmall}
+          label="Title"
+          value={title}
+          onChange={e => onChange({...value, title: e.target.value})}
+        />
+        <TextInput
+          marginBottom={Spacing.ExtraSmall}
+          label="Description"
+          value={description}
+          onChange={e => onChange({...value, description: e.target.value})}
+        />
+      </Box>
+    </>
+  )
+}
+
+export interface AddressInputProps extends MarginProps {
+  label?: string
+  description?: string
+  value: MarkerPoint[]
+  onChange(address?: MarkerPoint[]): void
+}
+
+export function AddressInput(props: AddressInputProps) {
+  return (
+    <AutocompleteInput
+      {...props}
+      label={props.label}
+      valueToChipData={markerPoint => ({
+        id: nanoid(),
+        label: markerPoint.address
+      })}>
+      {propsAutocompleteInput => <AddressInputList {...propsAutocompleteInput} />}
+    </AutocompleteInput>
+  )
+}
+
+function AddressInputList({
+  isOpen,
+  inputValue,
+  highlightedIndex,
+  getItemProps,
+  getMenuProps
+}: AutocompleteInputListProps) {
+  const [items, setItems] = useState<MarkerPoint[]>([])
+  const api = 'ab48a007c6dd4906adcd2871a5deaebe'
+  const fetchURL = `https://api.opencagedata.com/geocode/v1/json?key=${api}&q=${inputValue}&limit=5&pretty=1`
+
+  useEffect(() => {
+    async function fetchData() {
+      if (inputValue !== null && inputValue.length > 1) {
+        try {
+          const request = await axios.get(fetchURL)
```

Comment:
this logic should be in `utility.ts`

Reply:
And why did you use axios? The native fetch API can do the same.

Reply:
Done. 
I did use axios because I did run into troubles with the native fetch API and with axios it did work. Since I wanted to move forward and since axios seems to have a couple of [advantages over the native fetch API](https://medium.com/@jeffrey.allen.lewis/http-requests-compared-why-axios-is-better-than-node-fetch-more-secure-can-handle-errors-better-39fde869a4a6) I stuck with it.